license-maven-plugin-git/src/main/java/com/mycila/maven/plugin/license/git/GitLookup.java

```diff
@@ -58,6 +54,7 @@
   private final Repository repository;
   private final TimeZone timeZone;
   private final boolean shallow;
+  private final Set<String> commitsToIgnore;
```

Comment:
Why store as a `String` rather than as a `RevCommit`?  You can use `fromString()` to convert.  This buys you:
 - Case insensitivity on the hex strings since you're converting to numeric values
 - Type safety/self-documenting code
 - Automatic format checking (e.g., invalid commit id strings will throw IllegalArgumentException and let the user know which commit ID is problematic). Otherwise you'd fail silently.
 - Less space. The strings take 80 bytes, the `RevCommit` stores in a 20-byte array.
 - There's already an `equals()` based on the numeric value so you can compare the `RevCommit` directly (`commitsToIgnore.contains(commit)`) as you iterate rather than filling up the heap using `.name()` to create a temporary `String` for every commit, for every file, while iterating.

Reply:
Nice catch, I have changed it accordingly.